lib/room.cpp

```diff
@@ -2086,7 +2086,8 @@ RoomEventPtr makeReplaced(const RoomEvent& target,
                           const RoomMessageEvent& replacement)
 {
     auto originalJson = target.originalJsonObject();
-    originalJson[ContentKeyL] = replacement.contentJson();
+    originalJson[ContentKeyL] = replacement.originalJsonObject().
+            take(ContentKeyL).toObject().take("m.new_content"_ls).toObject();
```

Comment:
I see what you mean here but why sneak into the backdoor with `originalJsonObject()`? `replacement` is supposed to be `const` in this context.
```suggestion
    originalJson[ContentKeyL] = replacement.contentJson().value("m.new_content"_ls);
```

Reply:
I was lost in the woods. Thanks. :)